buku

```diff
@@ -2064,13 +2085,15 @@ class BukuDb:
                 return False
 
         # delete records in reverse order
+        ids = sorted(set(x[0] for x in results))
         with self.lock:
-            for pos, row in reversed(list(enumerate(results))):
-                self.delete_rec(row[0], delay_commit=True)
+            for pos, id in reversed(list(enumerate(ids))):
```

Comment:
There's no actual guarantee that the input list is sorted by index and contains unique entries; which led to a bunch of warnings about missing indices (likely accompanied by removing wrong records) when I was testing this on certain searches.